ex2_system/source/housekeeping/housekeeping_task.c

```diff
@@ -37,9 +37,14 @@ SAT_returnState start_housekeeping_daemon(void);
 static void *housekeeping_daemon(void *pvParameters) {
     TickType_t hk_delay = pdMS_TO_TICKS(1000);
     uint32_t seconds_delay = 30;
+    All_systems_housekeeping all_hk_data;
     for (;;) {
         // Call housekeeping and have them collect and store data to SD card
-        populate_and_store_hk_data();
+        //Uncomment the function below to mock all housekeeping data for testing purposes
+        //populate_and_store_hk_data();
+
+        //Call this function to populate all housekeeping data
+        collect_hk_from_devices(&all_hk_data);

```

Comment:
This won't work. populate_and_store_hk_data called the collect function but then it also stores that data to a file, which is very necessary.

Reply:
fixed!